lib/cocoapods/target/aggregate_target.rb

```diff
@@ -98,6 +98,7 @@ def merge_embedded_pod_targets(embedded_pod_targets_for_build_configuration)
                           target_definition, client_root, user_project, user_target_uuids, merged).tap do |aggregate_target|
         aggregate_target.search_paths_aggregate_targets.concat(search_paths_aggregate_targets).freeze
         aggregate_target.mark_application_extension_api_only if application_extension_api_only
+        aggregate_target.mark_build_library_for_distribution if build_library_for_distribution
```

Comment:
is there a test covering this?

Reply:
No, there isn't. I copied this from #9142 and there isn't a test for `mark_application_extension_api_only` neither (or I haven't found it), so I don't know how to test this.